spec/models/info_request_spec.rb

```diff
@@ -3327,4 +3327,31 @@ def email_and_raw_email(opts = {})
     end
   end
 
+  describe '#last_embargo_set_event' do
+
+    context 'if no embargo has been set' do
+      let(:info_request){ FactoryGirl.create(:info_request) }
+
+      it 'returns nil' do
+        expect(info_request.last_embargo_set_event).to be_nil
+      end
+
+    end
+
+    context 'if embargos have been set' do
+      let(:embargo){ FactoryGirl.create(:embargo) }
+      let(:embargo_extension){ FactoryGirl.create(:embargo_extension) }
+
+      it 'returns the last "set_embargo" event' do
+        embargo.extend(embargo_extension)
```

Comment:
Might be good to check we get the right event before we extend the embargo here too, just for completeness?